components/post_view/post_list/index.jsx

```diff
@@ -0,0 +1,448 @@
+        channelSyncStatus: PropTypes.bool,
+
+        /**
+         * Used for websocket connect change to trigger sync.
+         */
+        socketStatus: PropTypes.object,
+
+        /**
+         * Used for determining if we reached to the top of channel.
+         * Used for taking a backup before clearing from redux-store.
+         */
+        postIdsInCurrentChannel: PropTypes.array,
+
+        /*
+         * Object from react-router
+         */
+        match: PropTypes.shape({
+            params: PropTypes.shape({
+                messageId: PropTypes.string,
+            }).isRequired,
+        }).isRequired,
+
+        actions: PropTypes.shape({
+
+            /*
+             * Get post for permaview
+             */
+            getPostThread: PropTypes.func.isRequired,
+
+            /*
+             * Get unreads posts onload
+             */
+            loadUnreads: PropTypes.func.isRequired,
+
+            /*
+             * Get posts using BEFORE_ID and AFTER_ID
+             */
+            loadPosts: PropTypes.func.isRequired,
+
+            /*
+             * Used for clearing posts from channel
+             */
+            clearPostsFromChannel: PropTypes.func.isRequired,
+
+            /*
+             * Used for changing posts status of channel
+             * For notifying redux store when we hit bottom or top of posts
+             */
+            channelPostsStatus: PropTypes.func.isRequired,
+
+            /*
+             * Used for taking backup of postsInChannel before clearing so,
+             * we can use it incase of failure in loading posts
+             */
+            backUpPostsInChannel: PropTypes.func.isRequired,
+
+            /*
+             * Used for notifying redux store on channel sync complete
+             */
+            channelSyncCompleted: PropTypes.func.isRequired,
+
+            /*
+             * Used for adding back postIds to postsInChannel from backUpPostsInChannel
+             */
+            addPostIdsFromBackUp: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDoingInitialLoad: this.shouldLoadPosts(props),
+            newerPosts: {
+                loading: false,
+                allLoaded: (props.channelPostsStatus && props.channelPostsStatus.atEnd) || false,
+            },
+            olderPosts: {
+                loading: false,
+                allLoaded: this.oldestMessageLoadedInView(props),
+            },
+            lastViewedAt: props.lastViewedAt || props.member.last_viewed_at,
+        };
+    }
+
+    componentDidMount() {
+        if (this.state.isDoingInitialLoad) {
+            this.postsOnLoad(this.props.channel.id);
+        }
+        if (!this.props.channelSyncStatus && this.props.channelPostsStatus) {
+            this.syncPosts(this.props);
+        }
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
```

Comment:
Should we get rid of introducing `UNSAFE_componentWillReceiveProps`?

Reply:
Got a little lazy at the end of this so, I just added this instead of a new lifecycle method. Will change this.